pkg/log/fields.go

```diff
@@ -0,0 +1,91 @@
+package log
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+)
+
+func mergeFields(receiver map[string]interface{}, input map[string]interface{}) map[string]interface{} {
+	newMap := make(map[string]interface{}, len(receiver)+len(input))
+
+	for k, v := range receiver {
+		newMap[k] = prepareForLog(v)
+	}
+
+	for k, v := range input {
+		newMap[k] = prepareForLog(v)
+	}
+
+	return newMap
+}
+
+func prepareForLog(v interface{}) interface{} {
+	switch t := v.(type) {
+	case error:
+		// Otherwise errors are ignored by `encoding/json`
+		return t.Error()
+	case time.Time:
+		return v
+	case map[string]interface{}:
+		// perform a deep copy of any maps contained in this map element
+		// to ensure we own the object completely
```

Comment:
I'm not sure if I get your point right; but why would you want to have error fields inside a nested map?
Isn't a top-level (multi) error field enough, and anything nested might confuse you?

Reply:
In general yes, but if someone would do `map[string]interface{}{ "myError": err }`, that would be excluded from the JSON serializer, right? But this method here nicely converts them to strings by calling `Error`, so that should no longer a problem. Just thought we could also add that to the comment

Reply:
added a note to the comment